tempest/api/workloadmgr/workload_reassign/test_workload_reassign_same_project_same_user.py

```diff
@@ -0,0 +1,170 @@
+                LOG.error("Create user trust on project failed.")
+                raise Exception("Create user trust on project")
+
+            LOG.debug("Create workload")
+            workload_id = self.workload_create(
+                [vm_id], tvaultconf.parallel, workload_cleanup=True)
+            LOG.debug("Workload ID: " + str(workload_id))
+
+            if(workload_id is not None):
+                self.wait_for_workload_tobe_available(workload_id)
+                if(self.getWorkloadStatus(workload_id) == "available"):
+                    reporting.add_test_step("Create workload", tvaultconf.PASS)
+                else:
+                    LOG.error("Failed to get workload status as available for workload ID: " + str(workload_id))
+                    raise Exception("Create workload")
+            else:
+                LOG.error("Failed to create workload.")
+                raise Exception("Create workload")
+
+            #take a full snapshot of it.
+            snapshot_id = self.create_snapshot(workload_id, is_full=True)
+
+            self.wait_for_workload_tobe_available(workload_id)
+            snapshot_status = self.getSnapshotStatus(workload_id, snapshot_id)
+
+            if (snapshot_status == "available"):
+                LOG.debug("Full snapshot created.")
+                reporting.add_test_step("Create full snapshot", tvaultconf.PASS)
+                tests[0][1] = 1
+            else:
+                LOG.error("Full snapshot creation failed.")
+                raise Exception("Create full snapshot")
+
+
+            #assign tenant1 and user1 to workload created.
+            reporting.test_case_to_write()
+            reporting.add_test_script(tests[1][0])
+            LOG.debug("Assign tenant and user to workload ID: " + str(workload_id))
+
+            ### copy tenant1 and user1
+            tenant_id = CONF.identity.tenant_id
+            user_id = CONF.identity.user_id
+
+            LOG.debug(f"tenant_id {tenant_id} and user_id {user_id}")
+
+            rc = self.workload_reassign(tenant_id, workload_id, user_id)
+            if rc == 0:
+                LOG.debug("Workload reassign to tenant 1 passed")
+                reporting.add_test_step(
+                    "Workload reassign to tenant 1", tvaultconf.PASS)
+                tests[1][1] = 1
+            else:
+                LOG.error("Workload reassign to tenant 1 failed")
+                raise Exception("Workload reassign to tenant 1")
+
+
+            #assign tenant1 and user1 to the same workload again.
+            reporting.test_case_to_write()
+            reporting.add_test_script(tests[2][0])
+            LOG.debug("Reassign tenant-1 and user-1 to workload ID: " + str(workload_id))
+
+            #reassign the same user and project to the workload.
+            rc = self.workload_reassign(tenant_id, workload_id, user_id)
+            if rc == 0:
+                LOG.debug("Workload reassign from tenant 1 to tenant 1 passed")
```

Comment:
Is this code not duplicate of lines 127-135 ?

Reply:
It is dup but the log messages are different hence I kept those 5 lines as it is.